### PR TITLE
REL-2726 - P2 Checks for Nonsidereal 

### PR DESCRIPTION
**Pull request by tpolecat**
This adds four P2 checks:
- `NoSchedulingBlock` - no scheduling block for non-sidereal observation
- `SchedulingBlockOutsideSemester` - if scheduling block doesn't fall within program semester
- `NoEphemerisForSchedulingBlock` - if ephemeris is undefined at scheduling block start
- `NoHorizonsDesignation` - if a nonsidereal target has no HORIZONS designation (this means the high-res lookup will fail)

### CODE REVIEW

**Comment by cquiroz** [comment]
Looks fine :+1:

File: bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import scalaz._, Scalaz._
+
+final class NonSiderealTargetRules extends IRule {
+  import NonSiderealTargetRules._
+
+  implicit class SemesterOps(s: Semester) {
+    def contains(time: Long, site: Site): Boolean =
+      time >= s.getStartDate(site).getTime &&
+      time <=   s.getEndDate(site).getTime
+  }
```
Comment (swalker2m) [inline bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala]: Seems like maybe something that is useful enough to go into `Semester` itself.  It's surprising we've never needed it before.

File: bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          es.getObservationNode)
+    }
+
+  def checkNoEphemerisForSchedulingBlock(es: ObservationElements): IP2Problems =
+    new P2Problems <| { p2p =>
+      for {
+        ocn <- es.getTargetObsComponentNode.asScalaOpt.toList
+        toc <- es.getTargetObsComp.asScalaOpt.toList
+        tar <- toc.getTargetEnvironment.getTargets.asScalaList
+        nst <- tar.getNonSiderealTarget.toList if nst.horizonsDesignation.isEmpty
```
Comment (swalker2m) [inline bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala]: Couldn't a target have a horizons designation and yet not have ephemeris including the scheduling block start time?  That is, maybe we don't need the `if nst.horizonsDesignation.isEmpty` filter there or maybe we should but the condition should be reversed?

Comment (tpolecat) [inline bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala]: Yep, thanks. This was a copy/paste error. The condition shouldn't be there.

**Comment by swalker2m** [comment]
👍

**Pull request merged**